### DISCUSSION TITLE
Is there any way to whitelist the subnet ips at runtime.

### DISCUSSION THREAD

**Discussion by cheleswer** [svix/svix-webhooks#1072] [General]
I am aware that whitelisting of subnets can be done using the env variable, but that requires the pod restart to make that change available in svix-server. Is there any API or command that can be used to whitelist the subnets without restarting the SVIX-Server.

**Comment by tasn** [comment]
Hey @cheleswer, thanks for reaching out!

Not at the moment, but I'm curious to hear about your planned use-case, maybe this is something we can support. Why do you need dynamic filtering?

**Reply by cheleswer** [reply]
Use case is, SVIX service is running in customers closed environment and if they are using the webhooks to send data to their internal app, they would want to use the local subnet IPs which are not publicly accessible. Now webhooks are not something that everyone use from day one, so if customer decided to use this feature at later some point they have to add those whitelisting variable and restart the service. Restarting service, for whitelisting IP'S is not very much desirable, so if we can provide the ability to add those IPS using some API at runtime, we provide the ability to configure webhooks according to their network environment as and when needed.

**Reply by tasn** [reply]
Thanks for the info. The whitelisting feature should be thought of more as a firewall than something you need to modify every time you add an endpoint. So it's more tied to your network topology (that rarely changes) than anything else.

Additionally, the service can be restarted in a zero-downtime way (you can change the setting, start new pods, and only then shutdown the old ones), which makes me feel like it's even less of a concern.

So in short: it's a setting that should rarely be changed (even with your example), and when it does need changing it can be done with zero downtime. I don't think changing this to be dynamic is worth the complexity (and performance impact) required. Happy to be convinced otherwise, but we would probably not be implementing this at this time.

Thanks again for the feedback + suggestion!
